storage/innobase/handler/ha_innodb.cc

```diff
@@ -1820,6 +1820,9 @@ MYSQL_THD innobase_create_background_thd(const char* name)
 	MYSQL_THD thd= create_background_thd();
 	thd_proc_info(thd, name);
 	THDVAR(thd, background_thread) = true;
+#ifdef WITH_WSREP
+	wsrep_thd_off(thd);
+#endif /* WITH_WSREP */
```

Comment:
As far as I can tell, if we initialized this field in `create_background_thd()` (maybe we already do?), we would not need `wsrep_thd_off()` at all.

Reply:
There's no `create_background_thd()` in 10.4 and 10.5, where this fix is also required. As I added `wsrep_thd_off()` there, I ported it to 10.6 as well for consistency. Also, `create_background_thd()` is used not only in InnoDB, so I wanted to keep the fix local to InnoDB code and don't change higher-level code.

Reply:
Is this bug reproducible in 10.4 or 10.5? The locking during DDL operations was rewritten in 10.6.

I think that we need a minimal fix for 10.6 and later releases. If a fix with `wsrep_thd_off(THD*)` is really needed for 10.4 or 10.5, such a fix should be null-merged. As far as I can tell, e7549917e17d61ced4a99f5e4c168d780c02fab1 introduced `create_background_thd()` specifically for InnoDB. Which other code is using it? What harm would clearing the flag cause for other threads?

Reply:
The bug is not reproducible in 10.4 and 10.5, but I think the fix should be there as well as keeping wsrep flags in THD and trx_t in sync is the correct behavior.
I cannot find another usage of `create_background_thd()` in 10.6, but I think I saw it in more recent versions.
Since the function is in the higher-level server code, its usage may be extended at any time. I cannot reason about the consequences of clearing wsrep flag specifically for background threads, it seems ok, but usually these kind of fixes are done on the case-by-case basis.
Okay, I can remove `wsrep_thd_off()` from 10.6 upwards.